Upgrade bitnami/postgresql helm depenency from 10.5.3 to 12.1.9

snjypl opened the pull request:
Fixes:  https://github.com/apache/airflow/issues/27353

this is a continuation of  https://github.com/apache/airflow/pull/27357. i fixed the failing unittests


<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

potiuk reviewed (approved):
You can also add the original author as co-author

jedcunningham reviewed (approved):
I also verified the tarball came from bitnami.

snjypl committed:
Upgrade bitnami/postgresql helm depenency from 10.5.3 to 12.1.9

This version of the chart uses different variable names for setting usernames and passwords in the postgres database.
`postgresql.auth.enablePostgresUser` is used to determine if the "postgres" admin account will be created.
`postgresql.auth.postgresPassword` sets the password for the "postgres" user.
`postgresql.auth.username` and `postrgesql.auth.password` are used to set credentials for a non-admin account if desired.
`postgresql.postgresqlUsername` and `postgresql.postresqlPassword`, which were used in the previous version of the chart, are no longer used.
Users will need to change these variable names in their values files if they are using the helm chart.

Co-authored-by: Caleb Woofenden <caleb.woofenden@bitsighttech.com>
